src/provider/common/Injector.ts

```diff
@@ -70,7 +70,7 @@ export class Injector {
             if (typeof Bindings[key] === 'function') {
                 container.bind(Inject[key]).to(Bindings[key] as any).inSingletonScope();
 
-                if ((Bindings[key] as Function).prototype.hasOwnProperty('init')) {
+                if ((Bindings[key] as Function).prototype.hasOwnProperty('init') && global.hasOwnProperty('fin')) {
```

Comment:
Feel like I don't understand this? Only wait for dependency initialization if we're not in tests?

I feel like being able so set the bindings before this gets called, so this isn't necessary, might be tider

Reply:
I agree with Bryan on this front, but appreciate that we may need to just get this in. As a compromise I'd be fine with making and referencing a separate tech debt story to find a way to get this working properly.

Reply:
The `fin` guards have been removed and in place we have updated the injector to rebind certain injectables in the unit tests.  A primary example of such would be in the `AppDirectory.unittest.ts` file.